Emulator/BF.File.Emulator/BfEmulator.cs

```diff
@@ -0,0 +1,217 @@
+﻿using FileEmulationFramework.Interfaces.Reference;
+using FileEmulationFramework.Interfaces;
+using System.Collections.Concurrent;
+using System.Text;
+using BF.File.Emulator.Bf;
+using FileEmulationFramework.Lib.Memory;
+using BF.File.Emulator.Utilities;
+using FlowFormatVersion = AtlusScriptLibrary.FlowScriptLanguage.FormatVersion;
+using static BF.File.Emulator.Mod;
+using AtlusScriptLibrary.Common.Libraries;
+using AtlusScriptLibrary.Common.Text.Encodings;
+using AtlusScriptLibrary.Common.Logging;
+using Logger = FileEmulationFramework.Lib.Utilities.Logger;
+using System.Diagnostics;
+
+namespace BF.File.Emulator
+{
+    /// <summary>
+    /// Simple emulator for Atlus PAK files.
+    /// </summary>
+    public class BfEmulator : IEmulator
+    {
+        /// <summary>
+        /// If enabled, dumps newly emulated files.
+        /// </summary>
+        public bool DumpFiles { get; set; }
+
+        /// <summary>
+        /// Event that is fired when a stream is created, before redirection kicks in.
+        /// </summary>
+        public event Action<IntPtr, string, MemoryManagerStream>? OnStreamCreated;
```

Comment:
This event is unused; and should be removed along with all related code.

Reply:
Done in 58e9d4a